src/index.js

```diff
@@ -20,6 +20,31 @@ class Worker {
 		this.onerror = undefined;
 		this.onmessage = undefined;
 
+    // kill child fork if parent dies. (programming lingo has always
+    // seemed so macabre... slaves and masters, killing children, hm.)
+    {
+      // https://stackoverflow.com/questions/14031763/doing-a-cleanup-action-just-before-node-js-exits
+      function exitHandler(options, err) {
+          this.terminate();
+          if (options.cleanup) console.log('clean');
+          if (err) console.log(err.stack);
+          if (options.exit) process.exit();
+      }
+
+      //do something when app is closing
+      process.on('exit', exitHandler.bind(this,{cleanup:true}));
+
+      //catches ctrl+c event
+      process.on('SIGINT', exitHandler.bind(this, {exit:true}));
```

Comment:
Not sure about these three. `SIGINT` shouldn't necessarily spell unconditional death for workers, and `SIGUSR{1,2}` seem like their semantics should be decided by the user, not a utility module like this one.

I suspect hooking these would lead to surprising child deaths that would need to be debugged.